ui/pages/access-control/index.js

```diff
@@ -525,7 +539,13 @@ export default function AccessControl() {
           return g
         })
     )
-  }, [allGrants])
+
+    if (allGrants?.length === 0 && totalCount === 0) {
+      page > 1
+        ? router.replace(`/access-control?p=${page - 1}`)
```

Comment:
Could we send them back to the `totalPages - 1` from the list response rather than going back one page? 

This seems like it could be weird if you bookmarked a page like `/access-control?p=50`, then deleted a bunch of grants. It seems like this would iteratively go back with a lot of redirecting.

Reply:
oh yeah, this is a very good point.